core/domain/collection_domain.py

```diff
@@ -345,6 +346,72 @@ def from_dict(
 
         return collection
 
+    @classmethod
+    def deserialize(cls, json_string):
+        """Return a Collection domain object decoded from a JSON string.
+
+        Args:
+            json_string: str. A JSON-encoded string that can be
+                decoded into a dictionary representing a Collection. Only call
+                on strings returned from the platform.cache cache_services.
```

Comment:
Not sure what this last part means. I don't think this serialization/deserialization needs to be specific to memcache -- you can say "Only call on strings that were created using serialize()."